src/topology/monitor.rs

```diff
@@ -0,0 +1,314 @@
+use Error::{self, ArgumentError, OperationError};
+use Result;
+
+use bson::{self, Bson, oid};
+use chrono::{DateTime, UTC};
+
+use coll::options::FindOptions;
+use connstring::{self, Host};
+use cursor::Cursor;
+use pool::ConnectionPool;
+use wire_protocol::flags::OpQueryFlags;
+
+use std::collections::BTreeMap;
+use std::net::TcpStream;
+use std::sync::{Arc, RwLock};
+use std::sync::atomic::{AtomicBool, AtomicIsize, Ordering};
+use std::thread;
+
+use super::server::{ServerDescription, ServerType};
+use super::{DEFAULT_HEARTBEAT_FREQUENCY_MS, TopologyDescription};
+
+const DEFAULT_MAX_BSON_OBJECT_SIZE: i64 = 16 * 1024 * 1024;
+const DEFAULT_MAX_MESSAGE_SIZE_BYTES: i64 = 48000000;
+
+/// The result of an isMaster operation.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct IsMasterResult {
+    pub ok: bool,
+    pub is_master: bool,
+    pub max_bson_object_size: i64,
+    pub max_message_size_bytes: i64,
+    pub local_time: Option<DateTime<UTC>>,
+    pub min_wire_version: i64,
+    pub max_wire_version: i64,
+
+    // Shards
```

Comment:
Explain this further. I know that the `msg` field of an `isMaster` reply equals `isdbgrid` iff we are talking to a mongos. Make this clear.